README.md

```diff
@@ -86,12 +86,24 @@ that as an IAB-2.0 complaint download.  For segments, we wait for _all_ the byte
 to be downloaded before sending an IAB-2.0 complaint impression.
 
 Since we might receive additional requests _after we've logged a download_, we
-can end up firing a whole bunch of downloads for the same listener-session-digest.  
-So it's important that they're considered idempotent in BigQuery, or some sort of
-unique constraint is applied later on.
+also lock the impression via a redis hash:
 
-TODO: introduce some sort of "this-was-already-logged" redis lock so we don't fire
-a bunch of duplicates into BigQuery.
+```
+redis:6379> HGETALL dtcounts:imp:<listener-session>/<digest>
```

Comment:
Went with a hash because it seemed the most memory efficient?  Not entirely sure the 1-hour TTL is "right", but we'll find out.

Reply:
Disregarding storage costs, would using the 24 hour rolling window as TTL remove all chance of dupe logging here ?

Reply:
Unfortunately no.  So the "listener-session" is an md5 of listenerId + url + dateInUTC.  But it's the date of the redirect, not the date of the download.